orchagent/vxlanorch.cpp

```diff
@@ -2376,6 +2376,13 @@ bool EvpnRemoteVnip2pOrch::addOperation(const Request& request)
         return false;
     }
 
+    VRFOrch* vrf_orch = gDirectory.get<VRFOrch*>();
+    if (vrf_orch->isL3VniVlan(vni_id))
```

Comment:
1. wouldn't this check is applicable for fdborch also ? 
2. If the vni became an L3VNI after IMR routes were installed then OA will remove VLAN-VNI mappings and add the VRF-VNI mapping, leaving us with a scenario where the SAI has MAC/IMR routes with no corresponding vlan vni mappings. 
3. What happens without this fix ? Are there SAI errors we see ?

Reply:
1. In case of FDB orch we have check to ensure Vlan membership which will prevent FDB getting added since the IMR is prevented
2. Yes that's true. But as per the current behavior we will fail at removing VLAN VNI mapping at SAI level due to references. Do you think we should add a check at orchagent. I am not sure if this check is valid for other platforms and so I didn't prefer to add
3. Without the fix SAI returns error and orchagent crashes.